dagshub/common/cli.py

```diff
@@ -0,0 +1,73 @@
+import argparse
+import sys
+
+import click
+
+import dagshub.auth
+from dagshub.common import config
+
+parser = argparse.ArgumentParser(prog="dagshub")
+
+
+@click.group()
+@click.option("--host", default=config.host, help="Hostname of DagsHub instance")
+@click.pass_context
+def cli(ctx, host):
+    ctx.obj = {"host": host.strip("/")}
+
+
+@cli.command()
+@click.argument("project_root", default=".")
+@click.option("--repo_url", help="URL of the repo hosted on DagsHub")
+@click.option("--branch", help="Repository's branch")
+@click.option("--username", help="User's username")
+@click.option("--password", help="User's password")
+@click.option(
+    "--debug", default=False, type=bool, help="URL of the repo hosted on DagsHub"
+)
+@click.pass_context
+def mount(ctx, **kwargs):
+    """
+    Mount a DagsHub Storage folder via FUSE
+    """
+    # Since pyfuse can crash on init-time, import it here instead of up top
+    from dagshub.streaming import mount
+
+    if not kwargs["debug"]:
+        # Hide tracebacks of errors, display only error message
+        sys.tracebacklimit = 0
+    mount(**kwargs)
+
+
+@cli.group()
+@click.pass_context
+def auth(ctx):
+    """
+    Authentication commands
+    """
+    pass
+
+
+@auth.command(name="add")
+@click.argument("token")
+@click.pass_context
+def auth_add(ctx, token):
+    """
+    Add a long-lived auth token
+    """
+    dagshub.auth.add_app_token(token, ctx.obj["host"])
+    print("Token added")
```

Comment:
I like the `gcloud` / `docker` / `heroku` style:
`dagshub login` will actually trigger oauth. 
Beside that, maybe regular long-lived tokens shouldn't even be cached, just managed by the user with whatever secret system they want. They would need to pass it as an argument to whatever function they use.
For CLI mount that would need to be
```bash 
dagshub mount --username <user> --password <pass>
```
WDYT?

Reply:
Agree that `dagshub login` is probably better than `dagshub auth login`
Right now the CLI is way too small to care about making a "sensible" tree out of it in terms of verbs, so doing it the flat way is maybe a better idea.

re: passing secrets to the program, I actually completely forgot, but I also wanted to do an env var that will override the whole token logic, and `get_token()` would just return that.
From my chats with Guy, he said he wants there to be basically two flows:
- User just wants to try out DagsHub - let them use OAuth to auth as quickly as possible
- Advanced users should create a token and use that to auth in the client.
I don't really see a point in also keeping username/password auth tbh.